BLUESPAWN-client/src/mitigation/mitigations/MitigateM1028-WFW.cpp

```diff
@@ -0,0 +1,88 @@
+#include "mitigation/mitigations/MitigateM1028-WFW.h"
+#include "hunt/RegistryHunt.h"
+
+#include "util/configurations/Registry.h"
+#include "util/log/Log.h"
+
+using namespace Registry;
+
+namespace Mitigations {
+
+	MitigateM1028WFW::MitigateM1028WFW() :
+		Mitigation(
+			L"M1028-WFW - Windows Firewall must be enabled with no exceptions",
+			L"The Windows Firewall is an important host-based security control that"
+			" should not be disabled. Furthermore, it should not permit exceptions. "
+			"Instead, users should create proper rules for specific programs.",
+			L"wfw",
+			SoftwareAffected::OperatingSystem,
+			MitigationSeverity::High
+		) {}
+
+	bool MitigateM1028WFW::MitigationIsEnforced(SecurityLevel level) {
+		LOG_INFO("Checking for presence of " << name);
+
+		auto DomainProfile = RegistryKey{ HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Services\\SharedAccess\\Parameters\\FirewallPolicy\\DomainProfile" };
+		auto StandardProfile = RegistryKey{ HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Services\\SharedAccess\\Parameters\\FirewallPolicy\\StandardProfile" };
+		auto PublicProfile = RegistryKey{ HKEY_LOCAL_MACHINE, L"SYSTEM\\CurrentControlSet\\Services\\SharedAccess\\Parameters\\FirewallPolicy\\PublicProfile" };
+
+		for (auto key : { DomainProfile, StandardProfile, PublicProfile }) {
+			// V-17417
+			if (!key.ValueExists(L"EnableFirewall") || key.GetValue<DWORD>(L"EnableFirewall") != 1) {
+				LOG_VERBOSE(1, L"Value for EnableFirewall is not set to 1 for " << key.ToString());
+				return false;
+			}
+			if (!key.ValueExists(L"DisableNotifications") || key.GetValue<DWORD>(L"DisableNotifications") != 0) {
+				LOG_VERBOSE(1, L"Value for DisableNotifications is not set to 0 for " << key.ToString());
+				return false;
+			}
+			// V-17407
+			if (!key.ValueExists(L"DoNotAllowExceptions") || key.GetValue<DWORD>(L"DoNotAllowExceptions") != 1) {
+				LOG_VERBOSE(1, L"Value for DoNotAllowExceptions is not set to 1 for " << key.ToString());
```

Comment:
This value, unless I'm mistaken, applies the default inbound and outbound action to all connections, making inbound connections impossible to allow. This is often undesirable.